### PR TITLE
Add Carthage support

### PR DESCRIPTION
**Pull request by JonnyBeeGod**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -216,6 +217,13 @@ it, simply add the following line to your Podfile:
 pod "Seam3"
 ```
 
+Seam3 is also available through [Carthage](https://github.com/Carthage/Carthage). To install
+it, simply add the following line to your Cartfile:
+
+```
+github "JonnyBeeGod/Seam3"
```
Comment (paulw11) [inline README.md]: Please reference the main Seam3 repo, not your fork

**Review by paulw11 — CHANGES_REQUESTED** [review]
I don't use Carthage; Is there a need for specific Carthage examples?

File: README.md (hunk 1 above)
Comment (JonnyBeeGod) [reply README.md]: Correct, fixed. 

No need for specific carthage examples, see e.g. https://github.com/Alamofire/Alamofire#carthage 

One could add installation instructions for carthage like in the Alamofire repo but honestly I think this is out of scope

**Review by paulw11 — CHANGES_REQUESTED** [review]
The PR still contains files for Carthage example apps.

**Comment by JonnyBeeGod** [comment]
Can you clarify? Carthage needs a framework and therefor a *.xcodeproj so I reused the already existing *.xcodeproj from the Examples folder. Do you want me to create a new *.xcodeproj in the root directory?

**Comment by paulw11** [comment]
I have never used Carthage, so I am not clear what it needs, but I don't understand why the PR has xcodeproject files for the example apps; The example apps are not Seam3.  Seam3 is a framework.  The example apps just show how it can be used.  When I use cocoa pods to integrate Seam3 with a project I don't get any "example" files in my project.

**Comment by paulw11** [comment]
Ok, after some quick reading it looks like we need to create a new Xcode project in the directory above "Example" and create targets in that project for the framework on the three platforms.

**Comment by JonnyBeeGod** [comment]
okay, I moved the framework targets from the example Xcode project to a newly created xcodeproj in the root directory

File: Example/Seam3-Carthage-iOS/Seam3_Carthage_iOS.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+//
```
Comment (paulw11) [inline Example/Seam3-Carthage-iOS/Seam3_Carthage_iOS.h:1]: All of the changes to the files under the Example folder need to be removed from the PR.

File: Example/Seam3-Carthage-iOS/Seam3_Carthage_iOS.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+//  Seam3_Carthage_iOS.h
```
Comment (paulw11) [inline Example/Seam3-Carthage-iOS/Seam3_Carthage_iOS.h:2]: Revert this change

File: Example/Seam3-Carthage-watchOS/Info.plist [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (paulw11) [inline Example/Seam3-Carthage-watchOS/Info.plist:1]: Revert this change

File: Example/Seam3-Carthage-watchOS/Seam3_Carthage_watchOS.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
+//
```
Comment (paulw11) [inline Example/Seam3-Carthage-watchOS/Seam3_Carthage_watchOS.h:1]: Revert this change

File: Example/Seam3-Carthage/Info.plist [hunk 6]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (paulw11) [inline Example/Seam3-Carthage/Info.plist:1]: Revert this change

File: Example/Seam3-Carthage/Seam3_Carthage.h [hunk 7]
```diff
@@ -0,0 +1,19 @@
+//
```
Comment (paulw11) [inline Example/Seam3-Carthage/Seam3_Carthage.h:1]: Revert this change

File: Example/Seam3.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -36,6 +36,34 @@
 		BB9341621E5A52C000A0A63A /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = BB9341611E5A52C000A0A63A /* Assets.xcassets */; };
```
Comment (paulw11) [inline Example/Seam3.xcodeproj/project.pbxproj:36]: Revert this change

File: Example/Seam3.xcodeproj/xcshareddata/xcschemes/Seam3-Carthage-iOS.xcscheme [hunk 9]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (paulw11) [inline Example/Seam3.xcodeproj/xcshareddata/xcschemes/Seam3-Carthage-iOS.xcscheme:1]: Revert this change

File: Example/Seam3.xcodeproj/xcshareddata/xcschemes/Seam3-Carthage-iOS.xcscheme [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+<Scheme
```
Comment (paulw11) [inline Example/Seam3.xcodeproj/xcshareddata/xcschemes/Seam3-Carthage-iOS.xcscheme:2]: Revert this change

File: Example/Seam3.xcodeproj/xcshareddata/xcschemes/Seam3-Carthage-macOS.xcscheme [hunk 11]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (paulw11) [inline Example/Seam3.xcodeproj/xcshareddata/xcschemes/Seam3-Carthage-macOS.xcscheme:1]: Revert this change
